### PR TITLE
Add Unit Tests for Logs

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Added Unit Tests for Logs
- Update SamplingDecision enum values description to use Activity.Recorded instead of Activity.IsRecorded

### CODE REVIEW

File: test/OpenTelemetry.Tests/OpenTelemetry.Tests.csproj [hunk 1]
```diff
@@ -11,6 +11,15 @@
     <ProjectReference Include="..\..\src\OpenTelemetry.Exporter.Console\OpenTelemetry.Exporter.Console.csproj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp2.1'">
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.1.1" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.1.1" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1' Or '$(TargetFramework)' == 'net5.0'">
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/OpenTelemetry.Tests.csproj:19]: we are missing any .netframework targets?

File: test/OpenTelemetry.Tests/Logs/LogRecordTest.cs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        {
+            this.logger.LogInformation("Hello from {name} {price}.", "tomato", 2.99);
+            var state = this.exportedItems[0].State;
+            var itemCount = state.GetType().GetProperty("Count").GetValue(state);
+
+            // state has name, price and {OriginalFormat}
+            Assert.Equal(3, itemCount);
+
+            // Get value for {name}
+            var firstArgument = state.GetType().GetProperty("Item").GetValue(state, new object[] { 0 });
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Logs/LogRecordTest.cs:132]: we can avoid reflection here. Check the following:
```
var stateDictionary = (IReadOnlyCollection<KeyValuePair<string, object>>)logRecord.State;
foreach (var entry in stateDictionary )
{
}
```

Comment (utpilla) [reply test/OpenTelemetry.Tests/Logs/LogRecordTest.cs]: Fixed this,

File: test/OpenTelemetry.Tests/OpenTelemetry.Tests.csproj (hunk 1 above)
Comment (utpilla) [reply test/OpenTelemetry.Tests/OpenTelemetry.Tests.csproj]: Updated the check to exclude net452, net46, and netcoreapp2.1

File: test/OpenTelemetry.Tests/Logs/BatchExportLogRecordProcessorTest.cs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Logs;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Logs
+{
+    public class BatchExportLogRecordProcessorTest
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Logs/BatchExportLogRecordProcessorTest.cs:31]: Both Simple and Batch export processor are generic - we dont need to test it separately here. it should be tested outside of Logs.

File: test/OpenTelemetry.Tests/Logs/LogRecordTest.cs [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+        public void CheckStateForStructuredLogWithTemplate()
+        {
+            var message = "Hello from {name} {price}.";
+            this.logger.LogInformation(message, "tomato", 2.99);
+            var state = this.exportedItems[0].State as IReadOnlyList<KeyValuePair<string, object>>;
+
+            // state has name, price and {OriginalFormat}
+            Assert.Equal(3, state.Count);
+
+            // Check if first item is name
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Logs/LogRecordTest.cs:129]: Not sure if the order is guaranteed. We just need to validate that the name, price is present,  without considering its location.

File: test/OpenTelemetry.Tests/Logs/BatchExportLogRecordProcessorTest.cs (hunk 3 above)
Comment (utpilla) [reply test/OpenTelemetry.Tests/Logs/BatchExportLogRecordProcessorTest.cs]: Removed Simple and BatchExport Processor tests for Logs.

File: test/OpenTelemetry.Tests/Logs/LogRecordTest.cs (hunk 4 above)
Comment (utpilla) [reply test/OpenTelemetry.Tests/Logs/LogRecordTest.cs]: Fixed this.

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
Thanks for adding test coverage for Logs!

Okay to merge, no product change.

**Pull request merged**